pkg/cloudprovider/providers/fake/fake.go

```diff
@@ -92,7 +93,10 @@ func (f *FakeCloud) Clusters() (cloudprovider.Clusters, bool) {
 
 // ProviderName returns the cloud provider ID.
 func (f *FakeCloud) ProviderName() string {
-	return ProviderName
+	if f.Provider == "" {
```

Comment:
This is required for testing. I like your suggestion about `defaultProviderName`